### PR TITLE
Support reading BIGNUMERIC type in BigQuery

### PR DESCRIPTION
**Pull request by ebyhr**
- [x] Run all BigQuery tests

### CODE REVIEW

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java [hunk 1]
```diff
@@ -323,10 +324,12 @@ private static String quote(String value)
             return Optional.empty();
         }
         if (type instanceof DecimalType) {
+            String bigqueryTypeName = this.toString();
+            verify(bigqueryTypeName.equals("NUMERIC") || bigqueryTypeName.equals("BIGNUMERIC"), "Expect NUMERIC or BIGNUMERIC: %s", bigqueryTypeName);
```
Comment (hashhar) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java:328]:
```suggestion
            verify(bigqueryTypeName.equals("NUMERIC") || bigqueryTypeName.equals("BIGNUMERIC"), "Expected NUMERIC or BIGNUMERIC: %s", bigqueryTypeName);
```

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java [hunk 2]
```diff
@@ -335,6 +338,7 @@ public Type getNativeType(BigQueryType.Adaptor typeAdaptor)
     {
         switch (this) {
             case NUMERIC:
+            case BIGNUMERIC:
```
Comment (hashhar) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java:341]: Should there be a comment that BIGNUMERIC without precision and scale are filtered in `getColumns`?

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/Conversions.java [hunk 3]
```diff
@@ -64,8 +64,13 @@ public static ColumnMetadata toColumnMetadata(Field field)
                 .build();
     }
 
-    public static boolean isSupportedType(LegacySQLTypeName type)
+    public static boolean isSupportedType(Field field)
     {
+        LegacySQLTypeName type = field.getType();
+        // Skip BIGNUMERIC without parameters because the precision (77) and scale (38) is too large
+        if (type == LegacySQLTypeName.BIGNUMERIC && field.getPrecision() == null && field.getScale() == null) {
```
Comment (hashhar) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/Conversions.java:71]: Should we also skip when p > 38?

Or introduce the `DecimalModule` to allow mapping larger decimals using `decimal-mapping` config (see doc at https://trino.io/docs/current/connector/postgresql.html#decimal-type-handling)?

File: plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryTypeMapping.java [hunk 4]
```diff
@@ -156,6 +157,47 @@ public void testNumericMapping()
...
+                .addRoundTrip("BIGNUMERIC(1)", "BIGNUMERIC '-1'", createDecimalType(1, 0), "CAST(-1 AS DECIMAL(1, 0))")
+                .addRoundTrip("BIGNUMERIC(38)", "BIGNUMERIC '10000000002000000000300000000012345678'", createDecimalType(38, 0), "CAST('10000000002000000000300000000012345678' AS DECIMAL(38, 0))")
+                .addRoundTrip("BIGNUMERIC(38)", "BIGNUMERIC '-10000000002000000000300000000012345678'", createDecimalType(38, 0), "CAST('-10000000002000000000300000000012345678' AS DECIMAL(38, 0))")
+                .execute(getQueryRunner(), bigqueryCreateAndInsert("test.bignumeric"));
+    }
+
+    @Test
+    public void testUnsupportedBigNumericMapping()
+    {
+        try (TestTable table = new TestTable(getBigQuerySqlExecutor(), "test.unsupported_bignumeric", "(supported_column INT64, unsupported_column BIGNUMERIC)")) {
```
Comment (hashhar) [inline plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryTypeMapping.java:194]: Also test `BIGNUMERIC(40,2)` (p > 38)?

**Review by hashhar — COMMENTED** [review]
Some comments.

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/Conversions.java (hunk 3 above)
Comment (ebyhr) [reply plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/Conversions.java:71]: Thanks for catching that. I misunderstood the spec of BIGNUMERIC type. 
Filed #9923. I will add a logic to skip when `p > 38` in this PR.

**Review by hashhar — APPROVED** [review]
Looks good to me

**Pull request merged**